llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -789,9 +791,24 @@ int main(int argc, char **argv) {
   if (OutputFilename.getNumOccurrences() == 0)
     OutputFilename = (Twine(sys::path::stem(InputFilename)) + ".files").str();
 
+  TableFiles TblFiles = processOneModule(std::move(M));
   util::SimpleTable Table;
-  int Res = processOneModule(std::move(M), Table);
-  if (Res)
+
+  auto addTableColumn = [&Table, &TblFiles](std::string Str) {
+    auto &Files = TblFiles[Str];
+    if (Files.empty())
+      return 0;
+    Error Err = Table.addColumn(Str, Files);
+    CHECK_AND_EXIT(Err);
+    return 0;
+  };
+
+  int Res;
+  if ((Res = addTableColumn(COL_CODE)) != 0)
+    return Res;
+  if ((Res = addTableColumn(COL_PROPS)) != 0)
+    return Res;
+  if ((Res = addTableColumn(COL_SYM)) != 0)
     return Res;
```

Comment:
I'm also not sure that `addTableColumn` is necessary at all, because we could just use the following right here:
```c++
if (!TableFiles[COL_CODE].empty()) {
    CHECK_AND_EXIT(Table.addColumn(COL_CODE, TableFiles[COL_CODE]));
}
if (!TableFiles[COL_PROPS].empty()) {
    CHECK_AND_EXIT(Table.addColumn(COL_PROPS, TableFiles[COL_PROPS]));
}
if (!TableFiles[COL_SYM].empty()) {
    CHECK_AND_EXIT(Table.addColumn(COL_SYM, TableFiles[COL_SYM]));
}
```

Reply:
That is true. In the following patch (147eb5fa54a9eb5f5a376bb656911396d873d59a), the logic of `addTableColumn` becomes more complicated. So, let me keep it like it is.